src/VersionParser.php

```diff
@@ -21,16 +24,22 @@ class VersionParser
     private $versions;
 
     /**
+     * @var Composer\VersionParser
+     */
+    private $parser;
+
+    /**
      * @var string
      */
-    private $modifier = '[._-]?(?:(stable|beta|b|RC|alpha|a|patch|pl|p)(?:[.-]?(\d+))?)?([.-]?dev)?';
+    const GIT_DATA_MATCH = '/.*(-\d+-g[[:alnum:]]{7})$/';
 
     /**
      * @param array $versions
      */
     public function __construct(array $versions = array())
     {
         $this->versions = $versions;
+        $this->parser = new Parser; 
```

Comment:
It's not part of the PSR-2 standard which is enforced. We might adopt PSR-12 one day, if it ever leaves draft to get voted on, but I won't expect any contributor to follow a standard which we don't advertise/enforce upfront. That includes me and my own bias towards wondering why the seven hells I need to include parentheses when there are zero parameters ;).